### PR TITLE
JCF: bring codebase significantly closer into compliance with the DUN…

### PR DESCRIPTION
**Pull request by jcfreeman2**
…E C++ Style Guide

The header mostly says it all. These are changes which can't (at
currently) get picked up by the linter. Here are some of changes made:

* Error checking
I made an effort to ensure that every function which could throw an exception is wrapped in a try-catch, and every
function which returns a status variable has that variable checked. As I made these changes under a time
constraint, where I was unable to do this, I added TODOs.

* const-ification
Many variables and functions which were logically const but not declared as such are now const.
GraphConstructor::ConstructGraph, CommandFacility::listen, etc.

* Removal of unnecessary interface
E.g., replaced non-owning pointers with references when the address of an object isn't of interest

* Removal of raw owning pointers
...and replacement with smart pointers

* Formatting of TODOs
I retroactively added names, dates and emails to TODOs which were missing these

### CODE REVIEW

File: test/queue_IO_check.cxx [hunk 1]
```diff
@@ -52,12 +52,12 @@ auto timeout = std::chrono::milliseconds(100); ///< Queue's timeout
  */
 std::unique_ptr<dunedaq::appfwk::Queue<int>> queue = nullptr;
 
-constexpr int nelements = 100; ///< Number of elements to push to the Queue (total)
+int nelements = 10000000; ///< Number of elements to push to the Queue (total)
```
Comment (eflumerf) [inline test/queue_IO_check.cxx:55]: Is there a reason for this to be so high? This executable takes ~30 seconds to run now. Maybe 1,000,000?

File: src/QueueRegistry.cpp [hunk 2]
```diff
@@ -11,21 +11,18 @@
...
-
-QueueRegistry::~QueueRegistry() {}
+  std::unique_ptr<QueueRegistry> QueueRegistry::me_ = nullptr;
 
 QueueRegistry&
 QueueRegistry::get()
 {
   if (!me_) {
-    me_ = new QueueRegistry();
+    me_.reset( new QueueRegistry() );
```
Comment (eflumerf) [inline src/QueueRegistry.cpp:25]: One of the "modernize" guidelines recommends using std::make_unique in this situation.

File: test/FakeDataConsumerDAQModule.cpp [hunk 3]
```diff
@@ -28,6 +28,9 @@ namespace dunedaq::appfwk {
 FakeDataConsumerDAQModule::FakeDataConsumerDAQModule(const std::string& name)
   : DAQModule(name)
   , thread_(std::bind(&FakeDataConsumerDAQModule::do_work, this, std::placeholders::_1))
+  , nIntsPerVector_(999)
+  , starting_int_(-999)
+  , ending_int_(-999)
   , queueTimeout_(100)
   , inputQueue_(nullptr)
```
Comment (eflumerf) [inline test/FakeDataConsumerDAQModule.cpp:35]: Should we move initializers for basic-types to the header, using brace syntax?

File: test/FakeDataProducerDAQModule.cpp [hunk 4]
```diff
@@ -29,6 +29,11 @@ FakeDataProducerDAQModule::FakeDataProducerDAQModule(const std::string& name)
   , thread_(std::bind(&FakeDataProducerDAQModule::do_work, this, std::placeholders::_1))
   , outputQueue_(nullptr)
   , queueTimeout_(100)
+  , nIntsPerVector_(999)
+  , starting_int_(-999)
+  , ending_int_(-999)
+  , wait_between_sends_ms_(999)
```
Comment (eflumerf) [inline test/FakeDataProducerDAQModule.cpp:35]: Should we move initializers for concrete types to the header, using brace syntax?

File: test/FakeDataProducerDAQModule.cpp [hunk 5]
```diff
@@ -106,7 +111,11 @@ FakeDataProducerDAQModule::do_work(std::atomic<bool>& running_flag)
     oss.str("");
 
     TLOG(TLVL_TRACE) << get_name() << ": Pushing vector into outputQueue";
-    outputQueue_->push(std::move(output), queueTimeout_);
+    try {
+      outputQueue_->push(std::move(output), queueTimeout_);
+    } catch(const QueueTimeoutExpired& ex) {
+      TLOG(TLVL_TRACE) << "Push resulted in a timeout in FakeDataProducerDAQModule::do_work; data may be lost!";
```
Comment (eflumerf) [inline test/FakeDataProducerDAQModule.cpp:117]: Maybe ers::warning(ex) here?

File: test/queue_IO_check.cxx [hunk 6]
```diff
@@ -242,6 +246,14 @@ main(int argc, char* argv[])
     return 1;
   }
 
+  if (vm.count("nelements")) {
+    nelements = vm["nelements"].as<int>();
+
+    if (nelements <= 0) {
+      throw std::domain_error("# of elements must be a positive integer");
```
Comment (eflumerf) [inline test/queue_IO_check.cxx:253]: throw std::... is discouraged; we might want to concoct and throw an ERS Issue here

File: test/queue_IO_check.cxx (hunk 1 above)
Comment (eflumerf) [reply test/queue_IO_check.cxx:55]: Maybe a better comment to make would be to remove some of the defaults (queue_type?) and force the user to specify things via the command-line...But that could also be a separate Issue.

Comment (jcfreeman2) [reply test/queue_IO_check.cxx:55]: Happy to make it 1 million.

File: src/QueueRegistry.cpp (hunk 2 above)
Comment (eflumerf) [reply src/QueueRegistry.cpp:25]: Except that it doesn't build. Rescinding comment.

File: test/queue_IO_check.cxx (hunk 1 above)
Comment (jcfreeman2) [reply test/queue_IO_check.cxx]: Changed it to 1 million.

File: test/FakeDataConsumerDAQModule.cpp (hunk 3 above)
Comment (jcfreeman2) [reply test/FakeDataConsumerDAQModule.cpp]: After Slack discussion, agreed that parameters which are only ever meant to be settable via JSON should be set in the header.

File: test/FakeDataProducerDAQModule.cpp (hunk 4 above)
Comment (jcfreeman2) [reply test/FakeDataProducerDAQModule.cpp]: After Slack discussion, agreed that parameters which are only ever meant to be settable via JSON should be set in the header.

File: test/FakeDataProducerDAQModule.cpp (hunk 5 above)
Comment (jcfreeman2) [reply test/FakeDataProducerDAQModule.cpp]: Done.

File: test/queue_IO_check.cxx (hunk 6 above)
Comment (jcfreeman2) [reply test/queue_IO_check.cxx]: Replaced the throws of std::domain_error with 
```
ERS_DECLARE_ISSUE(appfwk,                 ///< Namespace                                                                  
                  ParameterDomainIssue,  ///< Issue class name                                                            
                  "ParameterDomainIssue: \"" << ers_messg << "\"",
                  ((std::string)ers_messg))
```

**Pull request merged**